lib/water_drop/config.rb

```diff
@@ -71,6 +71,12 @@ class Config
       #   delivery. Default is 100 messages. Disable buffer size based background deliveries by
       #   setting this to 0.
       setting :delivery_threshold, 100
+      # option [Boolean]
+      setting :idempotent, false
+      # option [Boolean]
+      setting :transactional, false
+      # option [Integer]
+      setting :transactional_timeout, 60
```

Comment:
We may need to do some backwards compatibility handling here. It looks like we depend on delivery boy `~> 0.2` which supported ruby-kafka `~> 0.4` which didn't have these configuration options.

Perhaps `ConfigApplier` can be updated to check if a configuration option exists before applying it?